### PR TITLE
Improve thread finishing

### PR DESCRIPTION
**Pull request by ahmetmircik**
Improvements based on the logs here: https://github.com/hazelcast/hazelcast/issues/17324

### CODE REVIEW

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2225/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2225/Hazelcast-pr-builder-2225.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.1:checkstyle (default) on project hazelcast: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There is 1 error reported by Checkstyle 8.29 with /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/spi/impl/executionservice/impl/ExecutionServiceImpl.java:342: 'if' construct must use '{}'s. [NeedBraces]
--------------------------
</details>
</pre>

File: hazelcast/src/main/java/com/hazelcast/cp/internal/datastructures/spi/blocking/AbstractBlockingService.java [hunk 1]
```diff
@@ -366,6 +367,9 @@ public void run() {
         @Override
         public void run() {
             for (Entry<CPGroupId, Collection<BiTuple<String, UUID>>> e : getWaitKeysToExpire().entrySet()) {
+                if (currentThread().isInterrupted()) {
+                    break;
+                }
                 tryReplicateExpiredWaitKeys(e.getKey(), e.getValue());
```
Comment (mdogan) [inline hazelcast/src/main/java/com/hazelcast/cp/internal/datastructures/spi/blocking/AbstractBlockingService.java:373]: If this thread is blocked inside `tryReplicateExpiredWaitKeys()` method, I don't think this change will help. I don't remember how `raftNode.replicate(op).get()` behaves when the local member is shutdown. We should investigate it too.

Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/cp/internal/datastructures/spi/blocking/AbstractBlockingService.java:380]: I have added timeout there as you suggested.

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2286/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2286/Hazelcast-pr-builder-2286.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.1:checkstyle (default) on project hazelcast: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There are 3 errors reported by Checkstyle 8.29 with /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/cp/internal/raft/impl/task/ReplicateTask.java:69:13: 'if' is not followed by whitespace. [WhitespaceAfter]
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/cp/internal/raft/impl/task/ReplicateTask.java:69:54: '{' is not preceded with whitespace. [WhitespaceAround]
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/main/java/com/hazelcast/cp/internal/raft/impl/task/ReplicateTask.java:71:40: '1000' is a magic number. [MagicNumber]
--------------------------
</details>
</pre>

**Review by mdogan — APPROVED** [review]

**Review by asimarslan — APPROVED** [review]
Client

**Review by blazember — APPROVED** [review]

**Pull request merged**